src/Interpreters/TreeRewriter.cpp

```diff
@@ -788,14 +788,21 @@ void expandOrderByAll(ASTSelectQuery * select_query)
     for (const auto & expr : select_query->select()->children)
     {
         if (auto * identifier = expr->as<ASTIdentifier>(); identifier != nullptr)
-            if (Poco::toUpper(identifier->name()) == "ALL" || Poco::toUpper(identifier->alias) == "ALL")
-                throw Exception(ErrorCodes::UNEXPECTED_EXPRESSION,
-                                "Cannot use ORDER BY ALL to sort a column with name 'all', please disable setting `enable_order_by_all` and try again");
-
+        {
+            if (identifier->alias.empty())
+            {
+                if (Poco::toUpper(identifier->name()) == "ALL")
+                    throw Exception(ErrorCodes::UNEXPECTED_EXPRESSION, "Cannot use ORDER BY ALL to sort a column with name 'all'");
```

Comment:
Maybe say how to fix it, e.g. "ORDER BY ALL is ambiguous because the SELECT list contains a column with name/alias 'all'. Use alias with a different name to clarify."

(same below, twice)